### PR TITLE
Fixes #31337 - Katello sends unauth. docker repos at proxy sync time

### PR DESCRIPTION
**Pull request by ianballou**
To be tested with https://github.com/Katello/smart_proxy_container_gateway/pull/2 if it's not already merged.  Use the same setup instructions.  Test instructions become simpler.  Just sync your smart proxy and curl the smart proxy api for the unauthenticated repo list: 
```
curl https://`hostname`/v2/unauthenticated_repository_list
```

### CODE REVIEW

File: lib/katello/engine.rb [hunk 1]
```diff
@@ -202,6 +202,7 @@ class Engine < ::Rails::Engine
       require "#{Katello::Engine.root}/app/services/katello/puppet_class_importer_extensions"
       require "#{Katello::Engine.root}/lib/proxy_api/pulp"
       require "#{Katello::Engine.root}/lib/proxy_api/pulp_node"
+      require "#{Katello::Engine.root}/lib/proxy_api/container_gateway"
```
Comment (jlsherrill) [inline lib/katello/engine.rb:205]: did you forget to add this file? :)

Comment (ianballou) [reply lib/katello/engine.rb:205]: I sure did! I'll add it in a second.

File: db/seeds.d/104-proxy.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def format_errors(model = nil)
     fail "Unable to create proxy feature: #{format_errors feature}"
   end
 
-  ["Pulp", "Pulp Node", "Pulpcore"].each do |input|
+  ["Pulp", "Pulp Node", "Pulpcore", "Container_Gateway"].each do |input|
```
Comment (ianballou) [inline db/seeds.d/104-proxy.rb:19]: I was hoping the feature would show up as "Container Gateway" without the underscore, but it seems the Foreman feature mapper adds the underscore.  It's not a huge deal, but if anyone knows a workaround without renaming the plugin let me know.

**Review by jjeffers — APPROVED** [review]
Tested this with previous smart proxy setup and eveything worked as expected. The only hiccup was a katello seed that didn't populate the smart proxy feature "ContainerGateway". After adding that via the console and refreshing the capsule, I was able to see the list of repos as per the testing instructions on the proxy via http/curl.

**Pull request merged**